internal/tools/log.go

```diff
@@ -0,0 +1,125 @@
+	LogQueryTool.Register(mcp)
+}
+
+type LogTag struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type LogQueryRequest struct {
+	ServiceID         string   `json:"service_id,omitempty"`
+	ServiceInstanceID string   `json:"service_instance_id,omitempty"`
+	EndpointID        string   `json:"endpoint_id,omitempty"`
+	TraceID           string   `json:"trace_id,omitempty"`
+	Tags              []LogTag `json:"tags,omitempty"`
+	Start             string   `json:"start,omitempty"`
+	End               string   `json:"end,omitempty"`
+	Step              string   `json:"step,omitempty"`
+	Cold              bool     `json:"cold,omitempty"`
+	PageNum           int      `json:"page_num,omitempty"`
+	PageSize          int      `json:"page_size,omitempty"`
+}
+
+// buildLogQueryCondition builds the log query condition from request parameters
+func buildLogQueryCondition(req *LogQueryRequest) *api.LogQueryCondition {
+	duration := BuildDuration(req.Start, req.End, req.Step, req.Cold, DefaultDuration)
+
+	var tags []*api.LogTag
+	for _, t := range req.Tags {
+		v := t.Value
+		tags = append(tags, &api.LogTag{Key: t.Key, Value: &v})
+	}
+
+	paging := BuildPagination(req.PageNum, req.PageSize)
+
+	cond := &api.LogQueryCondition{
+		ServiceID:         &req.ServiceID,
+		ServiceInstanceID: &req.ServiceInstanceID,
+		EndpointID:        &req.EndpointID,
+		RelatedTrace:      &api.TraceScopeCondition{TraceID: req.TraceID},
+		QueryDuration:     &duration,
+		Paging:            paging,
+	}
+
+	if len(tags) > 0 {
+		cond.Tags = tags
+	}
+	return cond
+}
+
+// queryLogs queries logs from SkyWalking OAP
+func queryLogs(ctx context.Context, req *LogQueryRequest) (*mcp.CallToolResult, error) {
+	cond := buildLogQueryCondition(req)
+
+	logs, err := swlog.Logs(ctx, cond)
+	if err != nil {
+		return mcp.NewToolResultError(fmt.Sprintf("failed to query logs: %v", err)), nil
+	}
+
+	jsonBytes, err := json.Marshal(logs)
+	if err != nil {
+		return mcp.NewToolResultError(fmt.Sprintf(ErrMarshalFailed, err)), nil
+	}
+	return mcp.NewToolResultText(string(jsonBytes)), nil
+}
+
+var LogQueryTool = NewTool[LogQueryRequest, *mcp.CallToolResult](
+	"query_logs",
+	"Query logs from SkyWalking OAP with flexible filters.\n\n"+
+		"Workflow:\n"+
+		"1. Use this tool to find logs matching specific criteria\n"+
+		"2. Specify one or more query conditions to narrow down results\n"+
+		"3. Use duration to limit the time range for the search\n"+
+		"4. Supports filtering by service, instance, endpoint, trace, tags, and time\n"+
+		"5. Supports cold storage query and pagination\n\n"+
+		"Examples:\n"+
+		"- {\"service_id\": \"Your_ApplicationName\", \"start\": \"2024-06-01 12:00:00\", "+
+		"\"end\": \"2024-06-01 13:00:00\"}: Query logs for a service in a time range\n"+
+		"- {\"trace_id\": \"abc123...\"}: Query logs related to a specific trace\n"+
+		"- {\"tags\": [{\"key\": \"level\", \"value\": \"ERROR\"}], \"cold\": true}: Query error logs from cold storage\n",
```

Comment:
consider using backtick string to avoid too many escape characters

https://go.dev/ref/spec#String_literals